nuheat/thermostat.py

```diff
@@ -174,41 +178,119 @@ def schedule_mode(self, mode):
 
         self.set_data({"ScheduleMode": mode})
 
+    @property
+    def hold_time(self):
+        """
+        Return a datetime for the current temporary hold time (or None)
+        """
+        if self._schedule_mode == config.SCHEDULE_TEMPORARY_HOLD:
+            return self._hold_time
+        else:
+            return None
+
+    @hold_time.setter
+    def hold_time(self, hold):
+        """
+        Set a temporary hold time (and change schedule_mode to Temporary Hold).
+
+        :param hold: datetime for temporary hold_time.
+        """
+        hold_gmt = hold.astimezone(timezone(timedelta(0), 'GMT'))
+
+        if hold_gmt < datetime.now(timezone.utc):
+            raise Exception("Invalid hold_time - must be in the future.")
+
+        hold_str = hold_gmt.strftime("%a, %d %b %Y %H:%M:%S %Z")
+        post_data = {
+            "ScheduleMode": config.SCHEDULE_TEMPORARY_HOLD,
+            "HoldSetPointDateTime": hold_str
+        }
+        self.set_data(post_data)
+
+    @property
+    def next_schedule_event(self):
+        """
+        Return dictionary containing information about the next scheduled
+        event.
+        """
+        if self._data is None:
+            return None
+
+        # Get thermostat's timezone from server data
+        tstat_tzs = self._data.get("TZOffset", "")
+        tstat_tz = time.fromisoformat("00:00:00" + tstat_tzs).tzinfo
+
+        # Convert now to thermostat's timezone, so today/tomorrow will be right
+        now = datetime.now().astimezone(tstat_tz)
+
+        # Find the first active event in the next week that is after now.
+        # Note: We start with yesterday (-1), in case yesterday's "Sleep" time
+        # is set for today.
+        for add_days in range(-1, 8):
+            today = now.date() + timedelta(days=add_days)
+            tomorrow = today + timedelta(days=1)
```

Comment:
since these dates aren't exactly "today" and "tomorrow" in the loop, the variable names might be more descriptive as `day` and `next_day`.

Reply:
Good call.  I think this code started out without the loop - so today/tomorrow made more sense at the time.  I'll make this change.